src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -728,17 +757,27 @@ def __call__(
 
         # 7. Denoising loop
         num_warmup_steps = len(timesteps) - num_inference_steps * self.scheduler.order
+
         with self.progress_bar(total=num_inference_steps) as progress_bar:
             for i, t in enumerate(timesteps):
                 # expand the latents if we are doing classifier free guidance
                 latent_model_input = torch.cat([latents] * 2) if do_classifier_free_guidance else latents
                 latent_model_input = self.scheduler.scale_model_input(latent_model_input, t)
 
+                # For classifier free guidance, we need to do two forward passes.
+                # Here we concatenate the unconditional and text embeddings into a single batch
+                # to avoid doing two forward passes
+                prompt_embeds_cond = (
```

Comment:
@patrickvonplaten 

we now allow dynamically change `guidance_scale` during the denoising process - I updated the code so that the `prompt_embeds`  will be computed at each step based on the `guidance_scale`.

otherwise user will have to change `prompt_embeds` in their callback function when they need to change `guidance_scale` - I image that's something easy to forget. 

let me know what you think

Reply:
Actually can we revert this as this increased the memory usage. Note that in case `guidance_scale` is set to 1.0 for one step it's ok if we still compute two prompt embeds - it's mathematically the same result.

Reply:
@patrickvonplaten 

> Note that in case guidance_scale is set to 1.0 for one step it's ok if we still compute two prompt embeds - it's mathematically the same result.

do you mean that we should keep the `latents` and `prompt_embeds` at same batch size (duplicated) even for the steps where `guidance_scale` is set to `1.0`? I think the whole point people want to zero out cfg after some steps is that they will be able to cut the batch size in half and reduce the inference time. 

I reverted the change i made here because it indeed increase memory usage. But I still wonder that if it makes sense to adjust the batch size of various tensor variables based on the `guidance_scale` value on our end, inside diffuser code. In a way that we won't increase memory usage of course.

The Alternative is to put all the relevant variables in `CALLBACK_ON_STEP_END_INPUTS` and let user to handle that in their callback functions. For stable diffusion pipeline I think only `prompt_embeds` could be adjusted.  for SDXL we have `add_text_embeds` and `add_time_ids`; controlnet also have an additional `image` variable .....

For now I will go with the alternative: I will start to apply the changes we made here to other pipelines and add all the variables that may be affected by `guidance_scale` into `CALLBACK_ON_STEP_END_INPUTS`. Let me know what yo think.